src/main/java/net/snowflake/ingest/streaming/internal/ParquetTypeGenerator.java

```diff
@@ -0,0 +1,290 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.*;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.apache.parquet.schema.Types;
+
+public class ParquetTypeGenerator {
```

Comment:
it looks the team code style in the client is to leave java doc comments for all class methods and fields :)